# Bonus: Temperature Analysis I

In [5]:
import pandas as pd
from datetime import datetime as dt

In [6]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [7]:
df.tail()

station        date  prcp  tobs
19545  USC00516128  2017-08-19  0.09    71
19546  USC00516128  2017-08-20   NaN    78
19547  USC00516128  2017-08-21  0.56    76
19548  USC00516128  2017-08-22  0.50    76
19549  USC00516128  2017-08-23  0.45    76

In [4]:
df.date.dtype

dtype('O')

In [5]:
# Convert the date column format from string to datetime
df.date = pd.to_datetime(df.date, infer_datetime_format=True)

In [6]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])

In [7]:
# Drop the date column
df = df.drop(columns='date')

In [8]:
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [9]:
from scipy import stats

In [10]:
# Filter data for desired months
june_data = df[df.index.month == 6]
december_data = df[df.index.month == 12]

In [11]:
# Identify the average temperature for June
june_data.mean()

C:\Users\tosun\AppData\Local\Temp/ipykernel_25372/4272335812.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  june_data.mean()


prcp     0.136360
tobs    74.944118
dtype: float64

In [12]:
# Identify the average temperature for December
december_data.mean()

C:\Users\tosun\AppData\Local\Temp/ipykernel_25372/3810961149.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  december_data.mean()


prcp     0.216819
tobs    71.041529
dtype: float64

In [13]:
# Create collections of temperature data
june_temp=june_data.tobs
december_temp=december_data.tobs

In [14]:

# Run paired t-test 
stats.ttest_ind(june_temp, december_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

* Run the unpaired t-test using ind in code and p value determines whether the difference in the mean if there is statistical mean. There is only 3.9 degrees Celsius, and the mean temperatures in June and December in all stations between 2010-2017. Unpaired T-test conducted, p-value is very low than standard threshold, so null hypothesis is rejected. The difference could be considered as signaficant.  